Runtime error with ibv_fork_init()

Intel CI reported the following failure:
```
21:07:54    CC       util/info.o
21:07:54    CCLD     util/fi_info
21:07:54  src/.libs/libfabric.so: undefined reference to `ibv_fork_init'
21:07:54  collect2: error: ld returned 1 exit status
21:07:54  make[1]: *** [util/fi_info] Error 1
21:07:54  make[1]: Leaving directory `/jenkinsbuild/workspace/ofi_libfabric_daily_master'
```
I believe this is coming from ofi_vrb_check_fork_enabled(), and may be the result of not linking libibverbs, but finding the header file.   The configure results are:

```
21:06:07  configure: efa provider: disabled
21:06:07  configure: *** Configuring usnic provider
21:06:07  configure: usnic provider: disabled
21:06:07  checking for pkg-config... /usr/bin/pkg-config
21:06:07  checking pkg-config is at least version 0.9.0... yes
21:06:07  configure: *** Configuring gni provider
21:06:07  checking for CRAY_GNI_HEADERS... no
21:06:07  configure: gni provider: disabled
21:06:07  configure: *** Configuring udp provider
21:06:07  checking for sys/socket.h... (cached) yes
21:06:07  configure: udp provider: build as plugin
21:06:07  configure: *** Configuring tcp provider
21:06:07  configure: tcp provider: build as plugin
21:06:07  configure: *** Configuring rxm provider
21:06:07  configure: rxm provider: include in libfabric
21:06:07  configure: *** Configuring mrail provider
21:06:07  configure: mrail provider: include in libfabric
21:06:07  configure: *** Configuring rxd provider
21:06:07  configure: rxd provider: include in libfabric
21:06:07  configure: *** Configuring bgq provider
21:06:07  checking for direct bgq provider... no
21:06:07  configure: bgq provider: disabled
21:06:07  configure: *** Configuring shm provider
21:06:07  checking for shm_open... (cached) no
21:06:07  checking sys/mman.h usability... yes
21:06:07  checking sys/mman.h presence... yes
21:06:07  checking for sys/mman.h... yes
21:06:07  looking for library without search path
21:06:07  checking for shm_open in -lrt... yes
21:06:07  configure: shm provider: build as plugin
21:06:07  configure: *** Configuring rstream provider
21:06:07  configure: rstream provider: disabled
21:06:07  configure: *** Configuring perf provider
21:06:07  configure: perf provider: disabled
21:06:07  configure: *** Configuring hook_debug provider
21:06:07  configure: hook_debug provider: disabled
21:06:07  checking that generated files are newer than configure... done
```
Neither EFA or verb providers are being included.

--- a/configure.ac
+++ b/configure.ac
@@ -465,8 +465,6 @@ AS_IF([test $have_uffd -eq 1],
 AC_DEFINE_UNQUOTED([HAVE_UFFD_UNMAP], [$have_uffd],
 	[Define to 1 if platform supports userfault fd unmap])
 
-AC_CHECK_HEADERS_ONCE(infiniband/verbs.h)
-
 dnl Check support to intercept syscalls
 AC_CHECK_HEADERS_ONCE(elf.h sys/auxv.h)
 

--- a/include/ofi.h
+++ b/include/ofi.h
@@ -374,8 +374,6 @@ static inline uint32_t ofi_xorshift_random_r(uint32_t *seed)
 
 uint32_t ofi_generate_seed(void);
 
-int ofi_vrb_check_fork_enabled(struct fid_domain *domain_fid);
-
 size_t ofi_vrb_speed(uint8_t speed, uint8_t width);
 
 #ifdef __cplusplus

--- a/prov/efa/src/efa_domain.c
+++ b/prov/efa/src/efa_domain.c
@@ -136,6 +136,47 @@ static int efa_open_device_by_name(struct efa_domain *domain, const char *name)
 	return ret;
 }
 
+/*
+ * Register a temporary buffer and call ibv_fork_init() to determine if fork
+ * support is enabled.
+ *
+ * This relies on internal behavior in rdma-core and is a temporary workaround.
+ */
+static int efa_check_fork_enabled(struct fid_domain *domain_fid)
+{
+	struct fid_mr *mr;
+	char *buf;
+	int ret;
+
+	buf = malloc(ofi_get_page_size());
+	if (!buf)
+		return -FI_ENOMEM;
+
+	ret = fi_mr_reg(domain_fid, buf, ofi_get_page_size(),
+			FI_SEND, 0, 0, 0, &mr, NULL);
+	if (ret) {
+		free(buf);
+		return ret;
+	}
+
+	/*
+	 * libibverbs maintains a global variable to determine if any
+	 * registrations have occurred before ibv_fork_init() is called.
+	 * EINVAL is returned if a memory region was registered before
+	 * ibv_fork_init() was called and returns 0 if fork support is
+	 * initialized already.
+	 */
+	ret = ibv_fork_init();
+
+	fi_close(&mr->fid);
+	free(buf);
+
+	if (ret == EINVAL)
+		return 0;
+
+	return 1;
+}
+
 static struct fi_ops efa_fid_ops = {
 	.size = sizeof(struct fi_ops),
 	.close = efa_domain_close,
@@ -240,7 +281,7 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 		efa_mr_cache_enable = 0;
 	}
 
-	if (efa_mr_cache_enable && ofi_vrb_check_fork_enabled(*domain_fid)) {
+	if (efa_mr_cache_enable && efa_check_fork_enabled(*domain_fid)) {
 		fprintf(stderr,
 		         "\nEnabling the memory registration cache and libibverbs "
 			 "fork support is currently not supported by the EFA\n"

--- a/src/common.c
+++ b/src/common.c
@@ -59,10 +59,6 @@
 #include <ifaddrs.h>
 #endif
 
-#if HAVE_INFINIBAND_VERBS_H
-#include <infiniband/verbs.h>
-#endif
-
 #include <ofi_signal.h>
 #include <rdma/providers/fi_prov.h>
 #include <rdma/fi_errno.h>
@@ -1739,51 +1735,6 @@ struct fid_nic *ofi_nic_dup(const struct fid_nic *nic)
 fail:
 	ofi_nic_close(&dup_nic->fid);
 	return NULL;
-}
-
-/*
- * Register a temporary buffer and call ibv_fork_init() to determine if fork
- * support is enabled.
- *
- * This relies on internal behavior in rdma-core and is a temporary workaround.
- */
-int ofi_vrb_check_fork_enabled(struct fid_domain *domain_fid)
-{
-#if HAVE_INFINIBAND_VERBS_H
-	struct fid_mr *mr;
-	char *buf;
-	int ret;
-
-	buf = malloc(ofi_get_page_size());
-	if (!buf)
-		return -FI_ENOMEM;
-
-	ret = fi_mr_reg(domain_fid, buf, ofi_get_page_size(),
-			FI_SEND, 0, 0, 0, &mr, NULL);
-	if (ret) {
-		free(buf);
-		return ret;
-	}
-
-	/*
-	 * libibverbs maintains a global variable to determine if any
-	 * registrations have occurred before ibv_fork_init() is called.
-	 * EINVAL is returned if a memory region was registered before
-	 * ibv_fork_init() was called and returns 0 if fork support is
-	 * initialized already.
-	 */
-	ret = ibv_fork_init();
-
-	fi_close(&mr->fid);
-	free(buf);
-
-	if (ret == EINVAL)
-		return 0;
-
-	return 1;
-#else
-	return 0;
-#endif
 }
 
 /*

